includes/Modules/Analytics_4.php

```diff
@@ -1150,7 +1150,7 @@ function ( $metric_def ) {
 
 						if ( is_string( $metric_def ) ) {
 							$metric->setName( $metric_def );
-						} elseif ( is_array( $metric_def ) && ! empty( $metric_def['name'] ) ) {
+						} elseif ( is_array( $metric_def ) ) {
```

Comment:
We can loosen these checks up a bit, to allow the new validation to occur and provide better feedback in case of invalid metrics.